compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java

```diff
@@ -238,9 +239,9 @@
 
     private Stack<BlockNode> blockNodeStack = new Stack<>();
 
-    private Stack<VariableNode> varStack = new Stack<>();
+    private Stack<BLangVariable> varStack = new Stack<>();
```

Comment:
Whats the reason for the type change here?

Reply:
This is to avoid unnecessary casting to BLangVariable in several places. And to avoid an additional iteration over varListStack to cast the variables to BLangVariable